lib/resty/upstream/healthcheck.lua

```diff
@@ -201,8 +199,14 @@ local function peer_ok(ctx, is_backup, id, peer)
     end
 end
 
+-- shortcut error function for check_peer()
+local function report_error(sock, ctx, is_backup, id, peer, ...)
+  if not peer.down then errlog(...) end
+  sock:close()
```

Comment:
It's called upon errors, in which case the socket needs closing. The one occasion that it is not needed is when the initial `connect` call fails. But imo that is not such a hot code path. 
But that is not my call, so let me know if you want me to change it.